### ISSUE TITLE
Wrong custom middleware documentation

### ISSUE THREAD

**Issue by HarrySky** [Kludex/starlette#663]
Writing middleware for #660 (Too Early) and stumbled upon an error when was following Middleware documentation:

`AttributeError: 'EarlyDataMiddleware' object has no attribute 'dispatch_func'`

I believe that `__init__` method in documentation:

```
class CustomHeaderMiddleware(BaseHTTPMiddleware):
    def __init__(self, app, header_value='Example'):
        self.app = app
        self.header_value = header_value
...
```

Should be used with super:

```
class CustomHeaderMiddleware(BaseHTTPMiddleware):
    def __init__(self, app, header_value='Example'):
        super().__init__(app)
        self.header_value = header_value
...
```

I can open a Pull Request that fixes documentation.

**Comment by lovelydinosaur** [comment]
Yup, your assessment looks correct to me, too.

**Comment by HarrySky** [comment]
Closed via #665
